File: Practica 1/S3/CryptoFilters/FilterManager.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   FilterManager.cpp
 * Author: reko98
 * 
 * Created on 25 de marzo de 2021, 12:55
 */

#include "FilterManager.h"

FilterManager::FilterManager() {
}

FilterManager::FilterManager(const FilterManager& orig) {
}

FilterManager::~FilterManager() {
}

void FilterManager::addFilter(string filterType, double filterValue){
	list.addFilter(filterType, filterValue);
}

void FilterManager::execution(CryptoCurrency& moneda){
	list.Execute(moneda);
}

File: Practica-2/App/crypto_visionary/lib/FilterManager.dart
import 'package:crypto_visionary/Cryptocurrency.dart';
import 'package:flutter/material.dart';
import 'Filter.dart';
import 'FilterChain.dart';

class FilterManager{
  FilterChain list = new FilterChain();

  void addFilter(Filter nuevo){
    list.addFilter(nuevo);
  }

  void execution(Cryptocurrency moneda){
    list.execute(moneda);
  }
}
